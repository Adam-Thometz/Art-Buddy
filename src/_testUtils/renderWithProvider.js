import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;