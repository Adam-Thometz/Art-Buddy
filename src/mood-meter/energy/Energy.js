import React from 'react';

import { useSelector } from 'react-redux';

import './Energy.css';

import Battery from './battery/Battery';

import prependBrightnessAndSaturation from '_utils/_general/prependBrightnessAndSaturation';

const GREEN_FILTER = prependBrightnessAndSaturation('invert(68%) sepia(56%) saturate(915%) hue-rotate(38deg) brightness(109%) contrast(92%)');
const YELLOW_FILTER = prependBrightnessAndSaturation('invert(99%) sepia(44%) saturate(4778%) hue-rotate(335deg) brightness(102%) contrast(104%)');
const RED_FILTER = prependBrightnessAndSaturation('invert(49%) sepia(82%) saturate(4390%) hue-rotate(339deg) brightness(101%) contrast(98%)');

const Energy = () => {
  const { energy } = useSelector(state => state.moodMeter);

  const batteries = [
    { width: '89%', id: 2, filter: GREEN_FILTER },
    { width: '48%', id: 1, filter: YELLOW_FILTER },
    { width: '12%', id: 0, filter: RED_FILTER },
  ];
  
  return (
    <section className='Energy'>
      <p>How is your energy level?</p>
      <div className='Energy-batteries'>
        {batteries.map(battery => (
          <Battery width={battery.width} id={battery.id} filter={battery.id === energy ? battery.filter : null} />
        ))}
      </div>
    </section>
  );
};

export default Energy;