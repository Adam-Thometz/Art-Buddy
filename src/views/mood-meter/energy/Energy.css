.Energy {
  display: flex;
  flex-direction: column;
}

.Energy-batteries {
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  width: 90%;
  margin: 0 auto;
}
