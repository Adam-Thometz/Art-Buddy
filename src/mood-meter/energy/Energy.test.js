import React from "react";

import renderWithProvider from '_testUtils/renderWithProvider';

import Energy from './Energy';

describe('Energy component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<Energy />);
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<Energy />);
    expect(asFragment()).toMatchSnapshot();
  });
});