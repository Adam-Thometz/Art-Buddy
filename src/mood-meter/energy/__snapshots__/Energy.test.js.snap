// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Energy component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="Energy"
  >
    <p>
      How is your energy level?
    </p>
    <div
      class="Energy-batteries"
    >
      <div
        class="Battery"
        data-testid="battery2"
        id="2"
      >
        <img
          alt=""
          id="2"
          src="file"
        />
        <img
          alt=""
          data-testid="bar"
          id="2"
          src="file"
          style="width: 89%;"
        />
      </div>
      <div
        class="Battery"
        data-testid="battery1"
        id="1"
      >
        <img
          alt=""
          id="1"
          src="file"
        />
        <img
          alt=""
          data-testid="bar"
          id="1"
          src="file"
          style="width: 48%;"
        />
      </div>
      <div
        class="Battery"
        data-testid="battery0"
        id="0"
      >
        <img
          alt=""
          id="0"
          src="file"
        />
        <img
          alt=""
          data-testid="bar"
          id="0"
          src="file"
          style="width: 12%;"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
